Release v1.0.0 (#2)
mattermost/mattermost-app-pagerduty: README.md

# Mattermost/PagerDuty Integration

* [Feature summary](#feature-summary)
* [Set up](#setting-up)
    * [Installation HTTP](#installation-http)
    * [Installation Mattermost Cloud](#installation-mattermost-cloud)
    * [Configuration](#configuration)
* [Admin guide](#admin-guide)
    * [Slash commands](#slash-commands)
* [End user guide](#end-user-guide)
    * [Get started](#get-started)
    * [Use /pd commands](#use-pd-commands)
* [Development environment](#development-environment)
    * [Manual installation](#manual-installation)
    * [Install dependencies](#install-dependencies)
    * [Run the local development environment](#run-the-local-development-environment)
    * [Run the local development environment with docker](#run-the-local-development-environment-with-docker)

> This application allows you to integrate PagerDuty to your Mattermost instance, letting you subscribe your PagerDuty's services on any custom channel to get notified about any update related to your services. Also, allows the user to create or display incidents, watch who is on-call, etc. without moving from the Mattermost window.

Reviewer: ```suggestion
This application allows you to integrate PagerDuty to your Mattermost instance, letting you subscribe your PagerDuty's services on any custom channel to be notified about any update related to your services. Also, allows the user to create or display incidents, watch who is on-call, etc. without leaving the Mattermost window.
```

# Feature summary

**PagerDuty to Mattermost notifications:** Link your Mattermost channels with the PagerDuty Service you want to track, so you and your team can get notifications about the creation and updates of each incident, with a user-friendly interface that will let you reassign, change priority, add notes, view details and more.

# Set up

## Installation HTTP

> To install, as a Mattermost system admin user, run the command ``/apps install http PAGERDUTY_API`` in any channel. The ``/pd`` command should be available after the configuration has been successfully installed.

Reviewer: ```suggestion
To install, as a Mattermost system admin user, run the command ``/apps install http PAGERDUTY_API`` in any channel. The ``/pd`` command should be available after the app has been successfully installed.
```

> The ``PAGERDUTY_API`` should be replaced with the URL where the Trello API instance is running. Example: ``/apps install http https://myapp.com/manifest.json``

Reviewer: ```suggestion
The ``PAGERDUTY_API`` should be replaced with the URL where this PagerDuty integration is running. Example: ``/apps install http https://myapp.com/manifest.json``
```

## Installation Mattermost Cloud

> To install, as a Mattermost system admin user, run the command ``/apps install listed pagerduty`` in any channel. The ``/pd`` command should be available after the configuration has been successfully installed.

Reviewer: Same comment as in other PRs

```suggestion
To install, as a Mattermost system admin user, open the Mattermost Marketplace and select **Install** located next to the PagerDuty app. The ``/pd`` command should be available after the integration has been successfully installed.
```

## Configuration

After [installing](#installation)) the app:

> 1. Open your PagerDuty profile to get your credentials and link to your Mattermost instance. 

Reviewer: ```suggestion
1. Open your PagerDuty profile to get your credentials, and follow the steps below to link to your Mattermost instance. 
```

> 2. Select the **Integration** tab in the toolbar. To the right of the popup (in the **Developer Tools** column) select **Developer Mode**

Reviewer: ```suggestion
2. Select the **Integration** tab in the toolbar. To the right of the popup (in the **Developer Tools** column) select **Developer Mode**.
```

3. On the **My Apps** page, select **Create New APP** and fill the form. 
4. When the form is completed, select **OAuth 2.0** in the same form (on the **Add** button below). 
5. In the **Redirect URL** field, add the following link (where "mattermostURL" should be replaced with the link to you Mattermost instance):
``{mattermostURL}/plugins/com.mattermost.apps/apps/pagerduty/oauth2/remote/complete``
6. Select **Save**.

> 7. The ``Client ID`` and ``Client Secret`` will be displayed. Save them in a safe place - the ``Client Secret`` can't be recovered). 

Reviewer: ```suggestion
7. The ``Client ID`` and ``Client Secret`` will be displayed. Temporarily copy them in order to use them in a step in the next section - the ``Client Secret`` can't be recovered). 
```

8. To finish, in the **Scopes** select **Read/Write**, then select **Save**.
9. Return to Mattermost. 
10. As a super admin role user, run the ``/pd configure`` command.
11. Enter the PagerDuty ``Client ID`` and ``Client Secret`` when prompted.

# Admin guide

## Slash commands

``/pd configure``: This command will enable all the other commands; it asks the administrator for a ``Client ID`` and ``Client Secret`` (which will be used to enable OAuth 2.0 authentication)

# End user guide

## Get started

## Use ``/pd`` commands
- ``/pd help``: This command will show all current commands available for this application.
- ``/pd connect``: Allow to any user to link their PagerDuty account with Mattermost. It must be executed to show the rest of the commands.
- ``/pd incident create``: Allow any user to create a new incident.
- ``/pd incident list``: Allow any user to display all the current incidents.
- ``/pd subscription add``: Allow any user to subscribe any channel to any PagerDuty service that the user has access.
- ``/pd subscription list``: Allow any user to show the current subscriptions for all the channels.

> - ``/pd subscription delete``: Will allow you to remove a subscription. No more notifications from that team will be received.

Reviewer: These descriptions should change after we change the subscription feature to be channel-aware, i.e. using `context.channel.id`
